components/apimgt/org.wso2.carbon.apimgt.keymgt/src/main/java/org/wso2/carbon/apimgt/keymgt/handlers/SessionDataPublisherImpl.java

```diff
@@ -0,0 +1,348 @@
+                                    + " : " + e);
+                        }
+                        if (scopedToken != null) {
+                            //Revoking token from database
+                            try {
+                                OAuthTokenPersistenceFactory.getInstance().getAccessTokenDAO()
+                                        .revokeAccessTokens(new String[] { scopedToken.getAccessToken() });
+
+                            } catch (IdentityOAuth2Exception e) {
+                                log.error("Error while revoking access tokens related for the application " + appDTO
+                                        .getApplicationName() + "and the for user " + authenticatedUser.getUserName()
+                                        + " : " + e);
+                            }
+                            //Revoking the oauth consent from database.
+                            try {
+                                OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO()
+                                        .revokeOAuthConsentByApplicationAndUser(
+                                                authzUser.getAuthenticatedSubjectIdentifier(), tenantDomain, username);
+                            } catch (IdentityOAuth2Exception e) {
+                                log.error("Error while revoking access tokens related for the application " + appDTO
+                                        .getApplicationName() + "and the for user " + authenticatedUser.getUserName()
+                                        + " : " + e);
+                            }
+                        }
+
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Method to retrieve applications authorized for user
+     * @param authenticatedUser authenticated user info
+     * @return array of authorized applications
+     * @throws IdentityOAuthAdminException exception
+     */
+    private OAuthConsumerAppDTO[] getAppsAuthorizedByUser(AuthenticatedUser authenticatedUser)
+            throws IdentityOAuthAdminException {
+
+        OAuthAppDAO appDAO = new OAuthAppDAO();
+
+        String tenantAwareusername = authenticatedUser.getUserName();
+        String tenantDomain = authenticatedUser.getTenantDomain();
+        String username = UserCoreUtil.addTenantDomainToEntry(tenantAwareusername, tenantDomain);
+
+        String userStoreDomain = authenticatedUser.getUserStoreDomain();
+
+        Set<String> clientIds;
+        try {
+            clientIds = OAuthTokenPersistenceFactory.getInstance().getTokenManagementDAO()
+                    .getAllTimeAuthorizedClientIds(authenticatedUser);
+        } catch (IdentityOAuth2Exception e) {
+            String errorMsg = "Error occurred while retrieving apps authorized by User ID : " + username;
+            throw handleError(errorMsg, e);
```

Comment:
Isn't the variable `errorMsg` redundant as it's used only once? Shall we replace the variable `errorMsg` in the `throw handleError` method call with the error message string?

Reply:
resolved